packages/block-library/src/page-list/edit.js

```diff
@@ -23,41 +23,185 @@ import {
 	Notice,
 	ComboboxControl,
 	Button,
+	Modal,
 } from '@wordpress/components';
 import { __, sprintf } from '@wordpress/i18n';
 import { useMemo, useState, useEffect } from '@wordpress/element';
 import { useEntityRecords } from '@wordpress/core-data';
-import { useSelect, useDispatch } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import ConvertToLinksModal from './convert-to-links-modal';
-import { convertToNavigationLinks } from './convert-to-navigation-links';
-import { convertDescription } from './constants';
+import { useConvertToNavigationLinks } from './use-convert-to-navigation-links';
 
 // We only show the edit option when page count is <= MAX_PAGE_COUNT
 // Performance of Navigation Links is not good past this value.
 const MAX_PAGE_COUNT = 100;
 const NOOP = () => {};
 
+const convertDescription = __(
+	'This menu is automatically kept in sync with pages on your site. You can manage the menu yourself by clicking customize below.'
+);
+
+function BlockContent( {
+	blockProps,
+	innerBlocksProps,
+	hasResolvedPages,
+	blockList,
+	pages,
+	parentPageID,
+} ) {
+	if ( ! hasResolvedPages ) {
+		return (
+			<div { ...blockProps }>
+				<Spinner />
+			</div>
+		);
+	}
+
+	if ( pages === null ) {
+		return (
+			<div { ...blockProps }>
+				<Notice status={ 'warning' } isDismissible={ false }>
+					{ __( 'Page List: Cannot retrieve Pages.' ) }
+				</Notice>
+			</div>
+		);
+	}
+
+	if ( pages.length === 0 ) {
+		return (
+			<div { ...blockProps }>
+				<Notice status={ 'info' } isDismissible={ false }>
+					{ __( 'Page List: Cannot retrieve Pages.' ) }
+				</Notice>
+			</div>
+		);
+	}
+
+	if ( blockList.length === 0 ) {
+		const parentPageDetails = pages.find(
+			( page ) => page.id === parentPageID
+		);
+		return (
+			<div { ...blockProps }>
+				<Warning>
+					{ sprintf(
+						// translators: %s: Page title.
+						__( '"%s" page has no children.' ),
+						parentPageDetails.title.rendered
+					) }
+				</Warning>
+			</div>
+		);
+	}
```

Comment:
Seems like there's a bug that can happen here. If I don't have a parent page defined, `parentPageDetails` ends up `undefined`, and an error can be thrown trying to access `parentPageDetails.title.rendered`.

Reply:
I can't reproduce it now, not sure what was happening. 🤔

Reply:
I found a different way to reproduce the issue, though it's a real edge case. I put together a fix - https://github.com/WordPress/gutenberg/pull/47603.

Looking a the diff, I realise the issue wasn't introduce here, so sorry for bombarding the PR with comments!